src/lib.rs

```diff
@@ -124,9 +120,13 @@ impl<'a> fmt::Display for Instr<'a> {
                         .join(", "),
                 )
             }
-            Self::Alloc4(size) => write!(f, "alloc4 {}", size),
-            Self::Alloc8(size) => write!(f, "alloc8 {}", size),
-            Self::Alloc16(size) => write!(f, "alloc16 {}", size),
+            Self::Alloc(size, align) => {
+                assert!(
+                    *align == 4 || *align == 8 || *align == 16,
+                    "Invalid stack alignment"
+                );
+                write!(f, "alloc{} {}", align, size)
```

Comment:
Not really a fan of this, as it moves the error from compile-time to run-time, which we should avoid.

Reply:
Not sure, we have to provide some way to get the corresponding allocation instruction by giving an alignment from `Type.align()` or similar

Reply:
But what keeps you from writing `qbe::Alloc(5, 0)`, which would result in a runtime error?

Reply:
An alternative would be to have `Alloc(Type, Align)` but that diverges too much from the original QBE IL. Let's just keep this as is.🙂